cmd/soroban-cli/src/fee.rs

```diff
@@ -16,15 +17,39 @@ pub struct Args {
     /// Number of instructions to simulate
     #[arg(long, help_heading = HEADING_RPC)]
     pub instructions: Option<u32>,
+    /// Build the transaction only write the base64 xdr to stdout
+    #[arg(long, help_heading = HEADING_RPC)]
+    pub build_only: bool,
+    /// Simulation the transaction only write the base64 to stdout
+    #[arg(long, help_heading = HEADING_RPC, conflicts_with = "build_only")]
+    pub sim_only: bool,
 }
 
 impl Args {
-    pub fn apply_to_assembled_txn(&self, txn: Assembled) -> Assembled {
-        if let Some(instructions) = self.instructions {
+    pub fn exit_if_build_only(&self, txn: &xdr::Transaction) -> Result<(), xdr::Error> {
+        if self.build_only {
+            println!("{}", txn.to_xdr_base64(xdr::Limits::none())?);
+            std::process::exit(0);
```

Comment:
Deep exits can be hard to track in code over time, is there a reason we can't `if build_only { return Ok(()) }` or similar in callers? It'd result in code flow in callers that's more obvious.